tokio-codec/src/lines_codec.rs

```diff
@@ -12,12 +12,93 @@ pub struct LinesCodec {
     // The next time `decode` is called with `abcde\n`, the method will
     // only look at `de\n` before returning.
     next_index: usize,
+
+    /// The maximum length for a given line. If `usize::MAX`, lines will be
+    /// read until a `\n` character is reached.
+    max_length: usize,
+
+    /// Are we currently discarding the remainder of a line which was over
+    /// the length limit?
+    is_discarding: bool,
```

Comment:
My initial expectation was that if a max length was exceeded, it'd just return an error and give up. This will try to keep reading and return the second line if under the max?

Reply:
It discards everything after `max_length` until it hits a newline, then resets.